deps/ncrypto/ncrypto.h

```diff
@@ -729,6 +731,28 @@ class CipherCtxPointer final {
   DeleteFnPtr<EVP_CIPHER_CTX, EVP_CIPHER_CTX_free> ctx_;
 };
 
+class KdfCtxPointer final {
+ public:
+  using ParamType =
+      std::variant<uint32_t, double, std::string, std::vector<char>>;
+  using Params = std::map<std::string, ParamType>;
```

Comment:
Maybe this can be a vector of pairs and we assume no one will add the same param twice (it will take the value of the first match)